### ISSUE TITLE
[HOLD on PR #34196] [$500] Profile -The preselected pronoun isn't highlighted by default

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#34749]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.27-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Open Expensify App
2. Log in
3. Navigate to settings by tapping on the user avatar
4. Tap on Profile and then Pronouns
5. Search and select a pronoun
6. Tap again Pronouns
7. Verify if the preselected pronoun is highlighted

## Expected Result:
The selected pronouns should have been highlighted by default when there is no searching performed (as in the timezone selection)

## Actual Result:
Only green checkmark exist in the preselected pronoun, but it is not highlighted (both if there is no search or after search)

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/68eefd9d-e152-47be-850a-ddd0dcdf3772

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01af844a0a64bbba29</li>
        <li>Upwork Job ID: 1748037184016420864</li>
        <li>Last Price Increase: 2024-01-25</li>
    </ul>
</details>

**Comment by esh-g** [comment]
# Proposal

## What is the root cause of this issue?
The root cause is that we don't recheck if `initallyFocusedOptionKey` is present after the first render. So after changing search text, we just focus to the `0` index and don't look for that `initallyFocusedOptionKey`

## What changes should be made to fix this?
We should modify this:
https://github.com/Expensify/App/blob/bcd783fd498fdd9112914d1c148e81b1495540e7/src/components/SelectionList/BaseSelectionList.js#L375-L378
To:
```js
// set the focus on the first item when the sections list is changed
if (sections.length > 0) {
    const initallyFocused = _.findIndex(flattenedSections.allOptions, (option) => option.keyForList === initiallyFocusedOptionKey)
    updateAndScrollToFocusedIndex(initallyFocused ?? 0);
}
```
This will check for the `initialItem` every time sections update.

**Comment by ikevin127** [comment]
Might be regression from https://github.com/Expensify/App/pull/30438.

A PR https://github.com/Expensify/App/pull/34196 is already opened that will tackle that regression. Another issue https://github.com/Expensify/App/issues/34300 that I was told will be fixed by the opened PR.

**Comment by DylanDylann** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The selected pronouns should have been highlighted by default **when there is no searching performed** (as in the timezone selection)

Currently the selected option isn't highlighted right after user go to pronouns page (don't type anything on search input)

<img width="880" alt="Screenshot 2024-01-19 at 15 04 52" src="https://github.com/Expensify/App/assets/141406735/7a31231b-d214-435d-8a61-4a656b680fa8">

In addition, When user type in the search input, It makes sense to focus on the first option

### What is the root cause of that problem?

Let's see here
https://github.com/Expensify/App/blob/991a4e213d6cb0af42b62b3e9f9c6b95b0317bea/src/components/SelectionList/BaseSelectionList.js#L158

We are setting the default value for `focusedIndex` in the param of useState hook, with this approach the default value only be calculated in the first render. But in the first render `flattenedSections.allOptions` is empty

<img width="442" alt="Screenshot 2024-01-19 at 15 08 55" src="https://github.com/Expensify/App/assets/141406735/376d1de2-f7df-4772-8aa6-c525a3a6e5f4">

Even though, after `flattenedSections.allOptions` is updated the `focusedIndex` will not be calculated again
### What changes do you think we should make in order to solve the problem?
We should use a useEffect to calculate `focusedIndex` again when `flattenedSections.allOptions` change
https://github.com/Expensify/App/blob/991a4e213d6cb0af42b62b3e9f9c6b95b0317bea/src/components/SelectionList/BaseSelectionList.js#L158

```
    useEffect(() => {
        const newFocusedIndex = _.findIndex(flattenedSections.allOptions, (option) => return option.keyForList === initiallyFocusedOptionKey)
        setFocusedIndex(newFocusedIndex)
    },[flattenedSections.allOptions, initiallyFocusedOptionKey])


```

### What alternative solutions did you explore? (Optional)
NA

### Result 

<img width="873" alt="Screenshot 2024-01-19 at 15 12 28" src="https://github.com/Expensify/App/assets/141406735/fd25d24b-8903-448f-b2cf-383646c85073">

**Comment by muttmuure** [comment]
Not overdue

**Comment by muttmuure** [comment]
Held

**Comment by muttmuure** [comment]
@fedirjh any closer on the linked PR?

**Comment by muttmuure** [comment]
Still waiting on the held PR

**Comment by muttmuure** [comment]
Yep, looks good
